src/boot/bootctl-status.c

```diff
@@ -503,6 +504,141 @@ int verb_status(int argc, char *argv[], void *userdata) {
         return r;
 }
 
+static int ref_file(Hashmap *known_files, const char *fn, const char *root)
+{
+        const char *p, *k = NULL;
+        int r, n;
+
+        assert(known_files);
+        if (!fn)
+                return -EINVAL;
+
+        p = prefix_roota(root, fn);
```

Comment:
hmm, I think the hashmap should contain the paths without "root" prefixed. dunno. would be simpler that way.

Reply:
Seems like it's possible that files with the same name  can exist on both the ESP as well as the XBOOTLDR partition. So have to store where the files are from to make sure we delete the correct one?

Reply:
but .conf files from the ESP may only reference resources from the ESP and XBOOTLDR .conf files only resources from XBOOTLDR

hence I think we should just run the whole algorithm twice: once for the ESP with the root set to the ESP and once on the XBOOTLDR partition with the root set to that. And the hashmap we track things in should be a fresh one on each invocation